typewriter/fixes/tests/test_util.py

```diff
@@ -0,0 +1,116 @@
+                         ("a", "w", "from x import b as y, c as z, a as w, d"),
+                         ("a", "z", "from x.b import a as z"),
+                         ("a", "z", "from x.b import b as y, c as w, a as z, d"),
+                         ("a", "z", "import b as x, c as y, a as z, d as w"))
+        for name, alias, import_ in passing_tests:
+            n = self.get_import_binding_info(import_, name, None)
+            self.assertTrue(n)
+            self.assertTrue(n.import_name == name)
+            self.assertTrue(n.binding_name == alias)
+
+    def test_in_function(self):
+        self.try_with("def foo():\n\tbar.baz()\n\tstart=3")
+
+
+class Test_find_import_binding_info_with_alias(TestCase):
+
+    def find_import_binding_info(self, name, string, package):
+        node = parse(string)
+        print(node)
+        # Return the node as an import binding if possible, None if otherwise
+        return fixer_utils.find_import_binding_info(name, node, package, return_if_aliased=True)
+
+    def try_with(self, string):
+        failing_tests = (("a", "a", "x", "from a import b as x"),
+                         ("a.d", "a", "x", "from a.d import b as x"),
+                         ("d.a", "a", "x", "from d.a import b as x"),
+                         (None, "a", "x", "import b as x"),
+                         (None, "a", "x", "import b as x, c, d as y"),
+                         ("x", "y", "a", "from x import y as a"),
+                         (None, "y", "b", "from x import y as a"))
+        for package, name, alias, import_ in failing_tests:
+            n = self.find_import_binding_info(name, import_ + '\n' + string, package)
+            if n:
+                # If valid binding, check to make sure import info is correct
+                self.assertFalse(n.binding_name == alias and n.import_name == name)
+            else:
+                # Should return None
+                self.assertFalse(n)
+
+        passing_tests = (("a", "z", "from x import a as z"),
+                         ("a", "z", "import a as z"),
+                         ("a.b", "z", "import a.b as z"),
+                         ("b", "z", "import a.b as z"),
+                         ("a", "z", "import a as z\nimport u as v"),
+                         ("a", "w", "from x import b as y, c as z, a as w, d"),
+                         ("a", "z", "from x.b import a as z"),
+                         ("a", "z", "from x.b import b as y, c as w, a as z, d"),
+                         ("a", "z", "import b as x, c as y, a as z, d as w"))
+        for name, alias, import_ in passing_tests:
+            n = self.find_import_binding_info(name, import_ + '\n' + string, None)
+            self.assertTrue(n)
+            # Because we aren't supporting the package argument yet we just need to check
+            # if the import name is the final name on the chain
+            self.assertTrue(n.import_name == name or n.import_name.split('.')[-1] == name)
```

Comment:
This implies that if we are looking for `bar`, that we will accept a returned result of `foo.bar`.  In what situations is this desired?  ie, if we have:

```
import foo.bar
from other import bar

bar.stuff()
```

...then we want to make sure we get the result from `other`, and not the one from `foo`, right?  ie, without the `from other import bar` line, an attempt to use just `bar` would error...